api/v1alpha1/mutation_types.go

```diff
@@ -19,14 +19,50 @@ type MutationSpec struct {
 	// +required
 	Source Source `json:"source"`
 
-	// +required
-	ConfigRef ConfigReference `json:"configRef"`
+	// +optional
+	ConfigRef *ConfigReference `json:"configRef,omitempty"`
 
 	// +required
 	SnapshotTemplate SnapshotTemplateSpec `json:"snapshotTemplate"`
 
 	// +optional
 	Values map[string]string `json:"values,omitempty"`
+
+	// +optional
+	PatchStrategicMerge *PatchStrategicMerge `json:"patchStrategicMerge,omitempty"`
+}
+
+// PatchStrategicMerge contains the source and target details required to perform a strategic merge
+type PatchStrategicMerge struct {
+	// +required
+	Source PatchStrategicMergeSource `json:"source"`
+
+	// +required
+	Target PatchStrategicMergeTarget `json:"target"`
+}
+
+// PatchStrategicMergeSource contains the details required to retrieve the source from a Flux source
+type PatchStrategicMergeSource struct {
+	SourceRef PatchStrategicMergeSourceRef `json:"sourceRef"`
+	Path      string                       `json:"path"`
+}
+
+// PatchStrategicMergeSourceRef contains the flux source identity metadata
+type PatchStrategicMergeSourceRef struct {
+	// +kubebuilder:validation:Enum=OCIRepository;GitRepository;Bucket
+	// +required
+	Kind string `json:"kind"`
+
+	// +required
+	Name string `json:"name"`
+
+	// +optional
+	Namespace string `json:"namespace,omitempty"`
```

Comment:
What is namespace in case it's empty?

Reply:
Thanks, good catch. It should default to same namespace as owner.

Reply:
Actually let's just make it required.